src/run/port.rs

```diff
@@ -170,12 +200,27 @@ impl Port {
     unsafe { self.lab().try_into().unwrap_unchecked() }
   }
 
-  /// Accesses the numeric value of this port; this is valid for [`Num`] ports.
+  /// Accesses the integer value of this port; this is valid for [`Int`] ports.
   #[inline(always)]
-  pub const fn num(&self) -> i64 {
+  pub const fn int(&self) -> i64 {
     self.0 as i64 >> 4
   }
 
+  /// Accesses the float value of this port; this is valid for [`F32`] ports.
+  #[inline(always)]
+  pub fn float(&self) -> f32 {
+    f32::from_bits((self.0 >> 4) as u32)
+  }
+
+  /// Accesses the numeric value of this port; this is valid for [`Int`] or
+  /// [`F32`] ports. This is meant for numeric operations to defer
+  /// interpreting this port as an integer or as a float until the operation
+  /// type is known.
+  #[inline(always)]
+  pub const fn num(&self) -> u64 {
+    self.0 >> 4
```

Comment:
hmm, this doesn't do sign extension; not sure if this is a problem, can you check?